styx-scheduler-service/src/test/java/com/spotify/styx/docker/KubernetesGCPServiceAccountSecretManagerTest.java

```diff
@@ -349,63 +371,6 @@ public void shouldHandleTooManyKeysCreated() throws IOException {
     sut.ensureServiceAccountKeySecret(WORKFLOW_ID.toString(), SERVICE_ACCOUNT);
   }
 
-  @Test
-  public void shouldHandlePermissionDeniedErrorsWhenDeletingServiceAccountKeys() throws Exception {
-    final Secret secret = fakeServiceAccountKeySecret(
-        SERVICE_ACCOUNT, SECRET_EPOCH, "json-key", "p12-key", EXPIRED_CREATION_TIMESTAMP.toString());
-
-    when(podList.getItems()).thenReturn(ImmutableList.of());
-    when(k8sClient.secrets()).thenReturn(secrets);
-    when(secrets.list()).thenReturn(secretList);
-    when(secretList.getItems()).thenReturn(ImmutableList.of(secret));
-
-    // Verify that the secret is delete even if we get permission denied errors on deleting the keys
-    final GoogleJsonResponseException permissionDenied = new GoogleJsonResponseException(
-        new HttpResponseException.Builder(403, "Forbidden", new HttpHeaders()),
-        new GoogleJsonError().set("status", "PERMISSION_DENIED"));
-    doThrow(permissionDenied).when(serviceAccountKeyManager).deleteKey(keyName(SERVICE_ACCOUNT, "json-key"));
-    doThrow(permissionDenied).when(serviceAccountKeyManager).deleteKey(keyName(SERVICE_ACCOUNT, "p12-key"));
-
-    sut.cleanup();
-
-    verify(serviceAccountKeyManager).deleteKey(keyName(SERVICE_ACCOUNT, "json-key"));
-    verify(serviceAccountKeyManager).deleteKey(keyName(SERVICE_ACCOUNT, "p12-key"));
-    verify(secrets).delete(secret);
-  }
-
-  @Test
-  public void shouldHandleErrorsWhenDeletingServiceAccountKeys() throws Exception {
```

Comment:
These two tests above were deleted since this class now relies on the `tryDeleteKey` method being the one handling possible errors in the key deletion process.